via/static/scripts/video_player/components/ToastMessages.tsx

```diff
@@ -0,0 +1,148 @@
+
+export type ToastMessage = {
+  id: string;
+  type: 'error' | 'success' | 'notice';
+  message: ComponentChildren;
+
+  /**
+   * Visually hidden messages are announced to screen readers but not visible.
+   * Defaults to false.
+   */
+  visuallyHidden?: boolean;
+
+  /**
+   * Determines if the toast message should be auto-dismissed.
+   * Defaults to true.
+   */
+  autoDismiss?: boolean;
+};
+
+type ToastMessageItemProps = {
+  message: ToastMessage;
+  onDismiss: (id: string) => void;
+};
+
+/**
+ * An individual toast message: a brief and transient success or error message.
+ * The message may be dismissed by clicking on it. `visuallyHidden` toast
+ * messages will not be visible but are still available to screen readers.
+ */
+function ToastMessageItem({ message, onDismiss }: ToastMessageItemProps) {
+  // Capitalize the message type for prepending; Don't prepend a message
+  // type for "notice" messages
+  const prefix =
+    message.type !== 'notice'
+      ? `${message.type.charAt(0).toUpperCase() + message.type.slice(1)}: `
+      : '';
+
+  return (
+    <Callout
+      classes={classnames({
+        'sr-only': message.visuallyHidden,
+      })}
+      status={message.type}
+      onClick={() => onDismiss(message.id)}
+      variant="raised"
+    >
+      <strong>{prefix}</strong>
+      {message.message}
+    </Callout>
+  );
+}
+
+export type ToastMessagesProps = {
+  messages: ToastMessage[];
+  onMessageDismiss: (id: string) => void;
+};
+
+/**
+ * A collection of toast messages. These are rendered within an `aria-live`
+ * region for accessibility with screen readers.
+ */
+export default function ToastMessages({
+  messages,
+  onMessageDismiss,
+}: ToastMessagesProps) {
+  const [dismissedMessages, setDismissedMessages] = useState<string[]>([]);
+  const scheduledMessages = useRef(new Set<string>());
+
+  const dismissMessage = useCallback(
+    (id: string) => {
+      if (dismissedMessages.includes(id)) {
+        return;
+      }
+
+      // Track that this message has been dismissed, and once animation ends,
+      // actually remove it
+      setDismissedMessages(ids => [...ids, id]);
+      setTimeout(() => onMessageDismiss(id), 500);
+    },
+    [onMessageDismiss, dismissedMessages]
+  );
+  const scheduleMessageDismiss = useCallback(
+    (id: string) => {
+      if (scheduledMessages.current.has(id)) {
+        return;
+      }
+
+      // Track that this message has been scheduled to be dismissed. After a
+      // period of time, actually dismiss it
+      scheduledMessages.current.add(id);
+      setTimeout(() => {
+        dismissMessage(id);
+        scheduledMessages.current.delete(id);
+      }, 5000);
+    },
+    [dismissMessage]
+  );
+
+  useEffect(() => {
+    messages.forEach(({ autoDismiss = true, id }) => {
+      if (autoDismiss) {
+        scheduleMessageDismiss(id);
+      }
+    });
+  }, [messages, scheduleMessageDismiss]);
```

Comment:
All these hooks/helpers allow to track which messages have been scheduled for dismissal (because they have `autoDismiss: true`) and which have already been dismissed and need to wait for the CSS animation before actually removing.

It is perhaps be too much logic for this component, but it's the only way to not leak implementation details. I'm mainly unhappy with the handling of auto-dismiss, but doing it from the outside of the component means the use of CSS animations needs to be known from the outside.